internal/schemautil/userconfig/util.go

```diff
@@ -174,3 +176,20 @@ func DecodeKey(k string) string {
 func IsKeyTyped(k string) bool {
 	return regexp.MustCompile(`^.*_(boolean|integer|number|string|array|object)$`).MatchString(k)
 }
+
+func UnmarshalUserConfig(src interface{}) (map[string]interface{}, error) {
```

Comment:
I think we can move `UnmarshalUserConfig` to the `internal/schemautiul` package and do not export this function `UnmarshalUserConfig -> unmarshalUserConfig` since we only use it there.